### PR TITLE
feature(k8s-eks): Add EKS backend

### PR DESCRIPTION
**Pull request by dkropachev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: defaults/aws_config.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 instance_provision: 'spot'
 spot_max_price: 0.60
 instance_provision_fallback_on_demand: false
-region_name:
-  - eu-west-1
+region_name: 'eu-west-1'
```
Comment (bentsi) [inline defaults/aws_config.yaml:4]: revert

File: sct.py [hunk 2]
```diff
@@ -296,6 +298,24 @@ def list_resources(ctx, user, test_id, get_all, get_all_running, verbose):
     else:
         click.secho("Nothing found for selected filters in GCE!", fg="yellow")
 
+    click.secho("Checking EKS...", fg='green')
+    gke_clusters = list_clusters_eks(tags_dict=params, verbose=verbose)
```
Comment (bentsi) [inline sct.py:302]: `gke_clusters` -> `eks_clusters`

File: vars/getJenkinsLabels.groovy [hunk 3]
```diff
@@ -17,6 +17,7 @@ def call(String backend, String aws_region=null) {
                           'gce': 'gce-sct-builders',
                           'k8s-gce-minikube': 'gce-sct-builders',
                           'k8s-gke': 'gce-sct-builders',
+                          'k8s-eks': 'aws-sct-builders-eu-west-1-new',
```
Comment (bentsi) [inline vars/getJenkinsLabels.groovy:20]: remember to remove when moving to SCT Runner infra

**Review by bentsi — COMMENTED** [review]
generalize YAML names (remove gke suffix)

File: sct.py (hunk 2 above)
Comment (dkropachev) [reply sct.py]: fixed

File: vars/getJenkinsLabels.groovy (hunk 3 above)
Comment (dkropachev) [reply vars/getJenkinsLabels.groovy]: removed

**Comment by dkropachev** [comment]
Successfull GKE run:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-3/120/

EKS run:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-5/126

File: defaults/aws_config.yaml (hunk 1 above)
Comment (dkropachev) [reply defaults/aws_config.yaml]: reverted

**Comment by dkropachev** [comment]
PR is ready, tested on GKE:
https://jenkins.scylladb.com/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-3/

and on EKS:
https://jenkins.scylladb.com/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-5/

**Comment by dkropachev** [comment]
PR has been rebased and retested:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-3/146
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-5/171

Pre-commit issue due to the pylint bug.

File: defaults/k8s_eks_config.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+availability_zone: 'a,b'
```
Comment (bentsi) [inline defaults/k8s_eks_config.yaml:1]: add a comment that it's needed for provision but the pool is actually created

**Review by bentsi — APPROVED** [review]

**Pull request merged**